### PR TITLE
Added a ILibraryManager based IAssemblyCatalog for DNX

### PR DESCRIPTION
**Pull request by thecodejunkie**
As principal as the `AppDomainAssemblyCatalog` but based around `ILibraryManager`

Will return the following assemblies

- The Nancy assembly
- Anything that has a reference on the Nancy assembly

### CODE REVIEW

File: src/Nancy/LibraryManagerAssemblyCatalog.cs [hunk 1]
```diff
@@ -0,0 +1,57 @@
+#if DNX
+namespace Nancy
+{
+    using System;
+    using System.Collections.Generic;
+    using System.Linq;
+    using System.Reflection;
+    using Microsoft.Extensions.PlatformAbstractions;
+
+    public class LibraryManagerAssemblyCatalog : IAssemblyCatalog
```
Comment (jchannon) [inline src/Nancy/LibraryManagerAssemblyCatalog.cs]: OMG WHERE IS THE XML?

File: src/Nancy/LibraryManagerAssemblyCatalog.cs [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+        private readonly AssemblyName nancyAssemblyName = typeof(LibraryManagerAssemblyCatalog).GetTypeInfo().Assembly.GetName();
+        private readonly Lazy<IReadOnlyCollection<Assembly>> assemblies;
+
+        public LibraryManagerAssemblyCatalog()
+        {
+            this.libraryManager = PlatformServices.Default.LibraryManager;
+            this.assemblies = new Lazy<IReadOnlyCollection<Assembly>>(this.LoadAllNancyReferencingAssemblies);
+        }
+
+        public IReadOnlyCollection<Assembly> GetAssemblies()
```
Comment (jchannon) [inline src/Nancy/LibraryManagerAssemblyCatalog.cs]: I HAVE NO CLUE WHAT THIS DOES WITHOUT XML :smile:

File: src/Nancy/LibraryManagerAssemblyCatalog.cs [hunk 3]
```diff
@@ -0,0 +1,65 @@
...
+    /// Default implementation of the <see cref="IAssemblyCatalog"/> interface, based on
+    /// retrieving <see cref="Assembly"/> information from an <see cref="ILibraryManager"/>.
+    /// </summary>
+    public class LibraryManagerAssemblyCatalog : IAssemblyCatalog
+    {
+        private readonly ILibraryManager libraryManager;
+        private readonly AssemblyName nancyAssemblyName = typeof(LibraryManagerAssemblyCatalog).GetTypeInfo().Assembly.GetName();
+        private readonly Lazy<IReadOnlyCollection<Assembly>> assemblies;
+
+        public LibraryManagerAssemblyCatalog()
```
Comment (jchannon) [inline src/Nancy/LibraryManagerAssemblyCatalog.cs]: XML

File: src/Nancy/LibraryManagerAssemblyCatalog.cs [hunk 4]
```diff
@@ -0,0 +1,65 @@
...
+        /// <summary>
+        /// Gets all <see cref="Assembly"/> instances in the catalog.
+        /// </summary>
+        /// <returns>An <see cref="IReadOnlyCollection{T}"/> of <see cref="Assembly"/> instances.</returns>
+        public IReadOnlyCollection<Assembly> GetAssemblies()
+        {
+            return this.assemblies.Value;
+        }
+
+        public IReadOnlyCollection<Assembly> LoadAllNancyReferencingAssemblies()
```
Comment (jchannon) [inline src/Nancy/LibraryManagerAssemblyCatalog.cs]: XLM or private?

File: test/Nancy.Tests/Unit/Bootstrapper/NancyInternalConfigurationFixture.cs [hunk 5]
```diff
@@ -12,7 +12,15 @@ public class NancyInternalConfigurationFixture
 
         public NancyInternalConfigurationFixture()
         {
-            this.typeCatalog = new DefaultTypeCatalog(new AppDomainAssemblyCatalog());
+            IAssemblyCatalog assemblyCatalog;
+
+#if !DNX
```
Comment (thecodejunkie) [inline test/Nancy.Tests/Unit/Bootstrapper/NancyInternalConfigurationFixture.cs]: I don't like these. It's done in 3 different test setups =/

Comment (jchannon) [inline test/Nancy.Tests/Unit/Bootstrapper/NancyInternalConfigurationFixture.cs]: not sure of the alternatives

**Comment by thecodejunkie** [comment]
Woho, all green ;)

File: test/Nancy.Tests/Unit/Bootstrapper/NancyInternalConfigurationFixture.cs (hunk 5 above)
Comment (thecodejunkie) [inline test/Nancy.Tests/Unit/Bootstrapper/NancyInternalConfigurationFixture.cs]: They'll do for now. Will have a revisit later

**Pull request merged**